File: frontend/src/resources/address.ts
import axios from "axios";
import { baseUrl } from "utils/requests";

export type Address = {
  id?: string;
  sequenceNumber?: number;
  localityId: string;
  number?: string;
  complement?: string;
  latitude?: number;
  longitude?: number;
  directLink?: string;
  active?: boolean;
};

class AddressService {
  async create(personId: string, address: Address): Promise<Address> {
    const response = await axios.post(`${baseUrl}/persons/${personId}/addresses`, address);
    return response.data;
  }
  async findAllByPerson(personId: string): Promise<Address[]> {
    const response = await axios.get(`${baseUrl}/persons/${personId}/addresses`);
    return response.data;
  }
}

export const useAddressService = () => new AddressService();

File: backend/src/main/java/com/pasifcode/syncdoc/domain/entity/Address.java
package com.pasifcode.syncdoc.domain.entity;

import jakarta.persistence.*;
import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name = "tb_address")
public class Address extends BaseEntity {

    @Column(nullable = false)
    private Long sequenceNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "person_id")
    private Person person;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "locality_id")
    private Locality locality;

    private String number;

    private String complement;

    private Double latitude;

    private Double longitude;

    @Column (length = 1000)
    private String directLink;

    private Boolean active;
}